include/ibamr/PhaseChangeHierarchyIntegrator.h

```diff
@@ -0,0 +1,511 @@
+
+    /*!
+     * \brief Register temperature variable \f$ T \f$.
+     */
+    void registerTemperatureVariable(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > T_var,
+                                     const bool output_T_var = true);
+
+    /*!
+     * \brief Reset cached hierarchy dependent data.
+     */
+    void
+    resetHierarchyConfigurationSpecialized(SAMRAI::tbox::Pointer<SAMRAI::hier::BasePatchHierarchy<NDIM> > hierarchy,
+                                           int coarsest_level,
+                                           int finest_level) override;
+
+    /*!
+     * \brief Set a grid function to provide initial conditions for \f$ \varphi \f$ variable.
+     */
+    void setLiquidFractionInitialCondition(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > lf_var,
+                                           SAMRAI::tbox::Pointer<IBTK::CartGridFunction> lf_init);
+    /*!
+     * \brief Set a grid function to provide initial conditions for  \f$ T \f$ variable.
+     */
+    void setTemperatureInitialCondition(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > T_var,
+                                        SAMRAI::tbox::Pointer<IBTK::CartGridFunction> T_init);
+
+    /*!
+     * \brief Set a grid function to provide initial conditions for \f$ \rho \f$ variable.
+     */
+    void setDensityInitialCondition(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > rho_var,
+                                    SAMRAI::tbox::Pointer<IBTK::CartGridFunction> rho_init);
+
+    /*!
+     * \brief Set boundary conditions for \f$ T \f$ variable.
+     */
+    void setTemperaturePhysicalBcCoef(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > T_var,
+                                      SAMRAI::solv::RobinBcCoefStrategy<NDIM>* T_bc_coef);
+    /*!
+     * \brief Return \f$ T \f$ boundary condition object.
+     */
+    SAMRAI::solv::RobinBcCoefStrategy<NDIM>* getTemperaturePhysicalBcCoef();
+
+    /*!
+     *  \brief Compute source terms of energy equation.
+     */
+    virtual void computeEnergyEquationSourceTerm(int F_scratch_idx, const double dt) = 0;
+
+    /*!
+     * \brief Get the solver of the energy equation.
+     */
+    SAMRAI::tbox::Pointer<IBTK::PoissonSolver>
+    getHelmholtzSolverForEnergyEquation(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > var);
+
+    /*!
+     * \brief Get the operator that is used to evaluate the right-hand side of energy equation.
+     */
+    SAMRAI::tbox::Pointer<IBTK::LaplaceOperator>
+    getHelmholtzRHSOperatorForEnergyEquation(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > var);
+
+    /*!
+     * \brief Register an IBTK::CartGridFunction object to specify the value of the
+     * source term for energy equation.
+     */
+    void setEnergyEquationSourceTermFunction(SAMRAI::tbox::Pointer<IBTK::CartGridFunction> F_fcn);
```

Comment:
Should we standardize on whether or not to abbreviate "function"? Both are used here.